subiquity/models/tests/test_subiquity.py

```diff
@@ -116,210 +118,250 @@ def assertConfigHasVal(self, config, path, val):
         confs = self.writtenFilesMatchingContaining(
-            config,
-            'etc/systemd/system/snapd.service.d/*.conf',
-            'HTTP_PROXY=')
+            config, "etc/systemd/system/snapd.service.d/*.conf", "HTTP_PROXY="
+        )
         self.assertTrue(len(confs) == 0)
 
     def test_keyboard(self):
         model = self.make_model()
         config = model.render()
-        self.assertConfigWritesFile(config, 'etc/default/keyboard')
+        self.assertConfigWritesFile(config, "etc/default/keyboard")
 
     def test_writes_machine_id_media_info(self):
         model_no_proxy = self.make_model()
         model_proxy = self.make_model()
-        model_proxy.proxy.proxy = 'http://something'
+        model_proxy.proxy.proxy = "http://something"
         for model in model_no_proxy, model_proxy:
             config = model.render()
-            self.assertConfigWritesFile(config, 'etc/machine-id')
-            self.assertConfigWritesFile(config, 'var/log/installer/media-info')
+            self.assertConfigWritesFile(config, "etc/machine-id")
+            self.assertConfigWritesFile(config, "var/log/installer/media-info")
 
     def test_storage_version(self):
         model = self.make_model()
         config = model.render()
-        self.assertConfigHasVal(config, 'storage.version', 1)
+        self.assertConfigHasVal(config, "storage.version", 1)
 
     def test_write_netplan(self):
         model = self.make_model()
         config = model.render()
         netplan_content = None
-        for fspec in config['write_files'].values():
-            if fspec['path'].startswith('etc/netplan'):
+        for fspec in config["write_files"].values():
+            if fspec["path"].startswith("etc/netplan"):
                 if netplan_content is not None:
-                    self.fail("writing two files to netplan?")
-                netplan_content = fspec['content']
+                    self.fail("writing two files for network config?")
+                netplan_content = fspec["content"]
+            if fspec["path"].endswith("cloud.cfg.d/90-installer-network.cfg"):
+                if netplan_content is not None:
+                    self.fail("writing two files for network config?")
+                netplan_content = fspec["content"]
         self.assertIsNot(netplan_content, None)
         netplan = yaml.safe_load(netplan_content)
-        self.assertConfigHasVal(netplan, 'network.version', 2)
+        self.assertConfigHasVal(netplan, "network.version", 2)
 
     def test_sources(self):
         model = self.make_model()
         config = model.render()
-        self.assertNotIn('sources', config)
+        self.assertNotIn("sources", config)
 
     def test_mirror(self):
         model = self.make_model()
-        mirror_val = 'http://my-mirror'
-        model.mirror.create_primary_candidate(mirror_val).elect()
+        mirror_val = "http://my-mirror"
+        model.mirror.set_mirror(mirror_val)
```

Comment:
Is this the part that tripped the tests?  A failed merge maybe?

Reply:
yes failed merge. I've rebased and have pushed the changeset now